op-plasma/commitment.go

```diff
@@ -45,31 +104,17 @@ func (c Keccak256Commitment) Verify(input []byte) error {
 	return nil
 }
 
-// Keccak256 creates a new commitment from the given input.
-func Keccak256(input []byte) Keccak256Commitment {
-	return Keccak256Commitment(crypto.Keccak256(input))
+// NewGenericCommitment creates a new commitment from the given input.
+func NewGenericCommitment(input []byte) GenericCommitment {
+	return GenericCommitment(input)
 }
 
-// DecodeKeccak256 validates and casts the commitment into a Keccak256Commitment.
-func DecodeKeccak256(commitment []byte) (Keccak256Commitment, error) {
+// DecodeGenericCommitment validates and casts the commitment into a GenericCommitment.
+func DecodeGenericCommitment(commitment []byte) (GenericCommitment, error) {
 	if len(commitment) == 0 {
 		return nil, ErrInvalidCommitment
 	}
-	if commitment[0] != byte(Keccak256CommitmentType) {
-		return nil, ErrInvalidCommitment
-	}
-	c := commitment[1:]
-	if len(c) != 32 {
-		return nil, ErrInvalidCommitment
-	}
-	return c, nil
-}
-
-// GenericCommitment is the default commitment type for op-plasma.
-type GenericCommitment []byte
-
-func NewGenericCommitment(commitment []byte) GenericCommitment {
-	return commitment
+	return commitment[:], nil
```

Comment:
This function moved upward slightly for more consistent ordering (New and Decode before struct functions), and the commitment data no longer strips the first byte because that is the responsibility of the Interface.